File: classical-components-demo/ehic-scanner/src/main/java/io/scanbot/example/EhicResultActivity.java
package io.scanbot.example;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import io.scanbot.hicscanner.model.HealthInsuranceCardField;
import io.scanbot.hicscanner.model.HealthInsuranceCardRecognitionResult;

public class EhicResultActivity extends AppCompatActivity {

    private static final String EXTRA_EHIC_RESULT = "EHIC_RESULT";

    public static Intent newIntent(final Context context, final HealthInsuranceCardRecognitionResult result) {
        final Intent intent = new Intent(context, EhicResultActivity.class);
        intent.putExtra(EXTRA_EHIC_RESULT, result);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_ehic_result);

        TextView surname = findViewById(R.id.surname);
        TextView givenName = findViewById(R.id.given_name);
        TextView dateOfBirth = findViewById(R.id.date_of_birth);
        TextView personalNumber = findViewById(R.id.personal_number);
        TextView institutionNumber = findViewById(R.id.institution_number);
        TextView institutionName = findViewById(R.id.institution_name);
        TextView cardNumber = findViewById(R.id.card_number);
        TextView cardDate = findViewById(R.id.card_date);

        final HealthInsuranceCardRecognitionResult result = getIntent().getParcelableExtra(EXTRA_EHIC_RESULT);

        for (HealthInsuranceCardField field : result.fields) {
            switch (field.type) {
                case SURNAME:
                    surname.setText(field.value);
                    break;
                case GIVEN_NAME:
                    givenName.setText(field.value);
                    break;
                case DATE_OF_BIRTH:
                    dateOfBirth.setText(field.value);
                    break;
                case PERSONAL_IDENTIFICATION_NUMBER:
                    personalNumber.setText(field.value);
                    break;
                case INSTITUTION_NUMBER:
                    institutionNumber.setText(field.value);
                    break;
                case INSTITUTION_NAME:
                    institutionName.setText(field.value);
                    break;
                case CARD_NUMBER:
                    cardNumber.setText(field.value);
                    break;
                case CARD_EXPIRATION_DATE:
                    cardDate.setText(field.value);
                    break;
            }
        }

        findViewById(R.id.retry).setOnClickListener(v -> finish());
    }
}

File: classical-components-demo/ehic-scanner/src/main/java/io/scanbot/example/EhicResultActivity.kt
package io.scanbot.example

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import io.scanbot.hicscanner.model.HealthInsuranceCardFieldType
import io.scanbot.hicscanner.model.HealthInsuranceCardRecognitionResult

class EhicResultActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ehic_result)

        val surname = findViewById<TextView>(R.id.surname)
        val givenName = findViewById<TextView>(R.id.given_name)
        val dateOfBirth = findViewById<TextView>(R.id.date_of_birth)
        val personalNumber = findViewById<TextView>(R.id.personal_number)
        val institutionNumber = findViewById<TextView>(R.id.institution_number)
        val institutionName = findViewById<TextView>(R.id.institution_name)
        val cardNumber = findViewById<TextView>(R.id.card_number)
        val cardDate = findViewById<TextView>(R.id.card_date)
        val country = findViewById<TextView>(R.id.country)
        val result : HealthInsuranceCardRecognitionResult = intent.getParcelableExtra(EXTRA_EHIC_RESULT)!!

        for ((type, value) in result.fields) {
            when (type) {
                HealthInsuranceCardFieldType.SURNAME -> surname.text = value
                HealthInsuranceCardFieldType.GIVEN_NAME -> givenName.text = value
                HealthInsuranceCardFieldType.DATE_OF_BIRTH -> dateOfBirth.text = value
                HealthInsuranceCardFieldType.PERSONAL_IDENTIFICATION_NUMBER -> personalNumber.text = value
                HealthInsuranceCardFieldType.INSTITUTION_NUMBER -> institutionNumber.text = value
                HealthInsuranceCardFieldType.INSTITUTION_NAME -> institutionName.text = value
                HealthInsuranceCardFieldType.CARD_NUMBER -> cardNumber.text = value
                HealthInsuranceCardFieldType.CARD_EXPIRATION_DATE -> cardDate.text = value
                HealthInsuranceCardFieldType.COUNTRY -> country.text = value
            }
        }
        findViewById<View>(R.id.retry).setOnClickListener { finish() }
    }

    companion object {
        private const val EXTRA_EHIC_RESULT = "EHIC_RESULT"
        @JvmStatic
        fun newIntent(context: Context?, result: HealthInsuranceCardRecognitionResult): Intent {
            val intent = Intent(context, EhicResultActivity::class.java)
            intent.putExtra(EXTRA_EHIC_RESULT, result)
            return intent
        }
    }
}